src/nixos/machines/tupac/config/disks.nix

```diff
@@ -0,0 +1,236 @@
+{ config, lib, ... }:
+
+# Nix-based Disk Management of TUPAC
+
+# Formatting strategy (Impermanence):
+#    Table: GPT
+#    2048 - 1050623 (1048576) [512M] -- EFI BOOT with FAT32
+#    1050624-1874579455 (1873528832) -- -60G Nix Store with BTRFS
+#    1874579456-2000408575 (125829120) -- Encrypted SWAP
+
+# Formatting strategy (WITHOUT impermanence):
+#    Table: GPT
+#    2048 - 1050623 (1048576) [512M] -- EFI BOOT with FAT32
+#    1050624-1874579455 (1873528832) -- -60G rootfs with BTRFS
+#    1874579456-2000408575 (125829120) -- Encrypted SWAP
+
+let
+	inherit (lib) mkMerge;
+in {
+	config = mkMerge [
+		{
+			age.secrets.tupac-disks-password.file = ../secrets/tupac-disks-password.age;
+
+			age.identityPaths = (if config.boot.impermanence.enable
+				then [ "/nix/persist/system/etc/ssh/ssh_host_ed25519_key" ]
+				else [ "/etc/ssh/ssh_host_ed25519_key" ]);
+		}
+
+		# FIXME-QA(Krey): Produces an infinite recursion -- (config.boot.impermanence.enable == true)
+		(if (true) then {
+			fileSystems."/nix/persist/system".neededForBoot = true;
+
+			# FIXME(Krey): Figure out how to do labels
+			disko.devices = {
+				nodev."/" = {
+					fsType = "tmpfs";
+					mountOptions = [
+						"size=2G"
+						"defaults"
+						# set mode to 755, otherwise systemd will set it to 777, which cause problems.
+						# relatime: Update inode access times relative to modify or change time.
+						"mode=755"
+					];
+				};
+
+				disk = {
+					system = {
+						device = "/dev/disk/by-id/nvme-SOLIDIGM_SSDPFKNU010TZ_BTEH24220RNQ1P0B"; # NVME SSD
+						type = "disk";
+						content = {
+							type = "gpt";
+							partitions = {
+
+								boot = {
+									type = "EF00"; # EFI System Partition/
+									start = "2048";
+									size = "512M";
```

Comment:
Although 512M is the standard recommended size for a boot partition, most of the space often remains unused. A size of around 250-300M is ideal, in my opinion

Reply:
> Although 512M is the standard recommended size for a boot partition, most of the space often remains unused. A size of around 250-300M is ideal, in my opinion -- @TanvirOnGH (https://github.com/NiXium-org/NiXium/pull/74#discussion_r1660035017)

250M was set previously infrastructure-wide for x86_64, but when i am testing a new feature which i usually do by deploying it to all system and 250M was kept holding me back for that as i usually need to go +-40 generations back sometimes etc..

Another rationale is that secure boot seems to make the generations bigger and i don't want to limit users for plymouth ricing.